### PR TITLE
Confirmation flow

### PR DESCRIPTION
**Pull request by adamdruppe**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -53,6 +53,68 @@ def check_credentials
...
+  def prep_confirm_campaign_info
+    # These should never be nil at this point, so I'm not checking
+    # it - if it does happen, the error report should help us find
+    # why it was nil and fix that root cause bug.
+
+    # Both @enrollment and campaign should never be nil.
+
+    @enrollment = Enrollment.find_by(:user_id => current_user.id)
+
+    if false # hack
```
Comment (sadleb) [inline app/controllers/users_controller.rb]: If you're going to leave this code here, please update the ```# hack``` comment to tell how to use this for testing.

Comment (adamdruppe) [inline app/controllers/users_controller.rb]: Oh this PR isn't quite ready yet, a few more changes are
coming today based on a round of testing we did on dev
last night.

File: app/views/admin/users/bulk_student_upload.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<h1>Bulk Student Registration Upload</h1>
+
+<p>This will let you upload student IDs to Salesforce. You still need to Sync to LMS on SF for them to get into Canvas; this just sets the registration codes.</p>
+
+<table class="csv-sample" width="600"><caption>Expected format</caption>
+  <thead><tr>
+  </tr></thead>
+  <tbody>
+    <tr>
```
Comment (sadleb) [inline app/views/admin/users/bulk_student_upload.html.erb]: Expected format == nil ? :)

File: app/views/users/confirm_part_2.html.erb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  <div class="plank plain-lt">
+    <div class="section-container">
+      <section>
+        <div class="col-sm-4 col-sm-offset-4">
+          <% if @selected_time %>
+            <p>You selected <strong><%= @selected_time %></strong>. Sound right? Click Submit to make it official!</p>
+            <input type="hidden" name="selected_time" value="<%@selected_time%>" />
+          <% else %>
+            <div class="availability-options">
+              <p>Drat. We'll do what we can to include you in the program. In the meantime, we'll add you to our waitlist. Just let us know which of these currently closed section meeting times work for you:</p>
```
Comment (sadleb) [inline app/views/users/confirm_part_2.html.erb]: Love the voice here.

File: app/views/users/confirm_part_2.html.erb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+              <% @times.each do |time| %>
+                <% if time['slots'] == 0 %>
+                  <label><input onchange="if(this.checked) document.getElementById('none').checked = false;" type="checkbox" class="some-time-option" name="times[]" /> <%= time['time'] %></label><br />
+                <% end %>
+              <% end %>
+              <label><input id="none" type="checkbox" name="none" onchange="if(this.checked) { var i; var a = document.querySelectorAll('.some-time-option'); for(i = 0; i < a.length; i++) a[i].checked = false;}" /> I cannot coach at any of these times either.</label>
```
Comment (sadleb) [inline app/views/users/confirm_part_2.html.erb]: This should probably happen on submit.  If I accidentally check it and it clears my selection, that's annoying.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (sadleb) [inline app/controllers/users_controller.rb]: > coming today based on a round of testing we did on dev
> last night.

Sounds good.  Let me know.

**Pull request merged**